syntax = "proto3";

package kagami_supervisor;

service Supervisor {
    rpc worker_report_in (WorkerReportInRequest) returns (WorkerReportInRequest);
    rpc update_provider_status (UpdateProviderRequest) returns (UpdateProviderStatusResponse);
}

message WorkerReportInResponse {
    string supervisor_addr = 1;
}

message UpdateProviderStatusResponse {
    int32 provider_id = 1;
}

message WorkerReportInRequest {
    string worker_addr = 1;
}

message RegisterResponse {
    bool accepted = 1;
}

message UpdateProviderRequest {
    string worker_addr = 1;
    int32 provider_replica_id = 2;
    int32 provider_status = 3;
}
